internal/controller/controllers/agent_controller.go

```diff
@@ -905,6 +914,17 @@ func (r *AgentReconciler) updateIfNeeded(ctx context.Context, log logrus.FieldLo
 	return nil
 }
 
+func (r *AgentReconciler) getIgnitionToken(ctx context.Context, log logrus.FieldLogger, ignitionEndpointTokenReference *aiv1beta1.IgnitionEndpointTokenReference) (string, error) {
+	secret := &corev1.Secret{}
+	secretRef := types.NamespacedName{Namespace: ignitionEndpointTokenReference.Namespace, Name: ignitionEndpointTokenReference.Name}
+	if err := r.Get(ctx, secretRef, secret); err != nil {
+		return "", errors.Wrap(err, fmt.Sprintf("Failed to get user-data secret %s/%s", ignitionEndpointTokenReference.Namespace, ignitionEndpointTokenReference.Name))
+	}
+
+	token := string(secret.Data["value"])
```

Comment:
Need to verify that Data contains the key:
	data, ok := secret.Data["value"]
	if !ok {
		return "", errors.Errorf("secret %s did not contain key value", ref.Name,)
	}

Reply:
Also maybe have a more meaningful name for the key

Reply:
Done